src/app.rs

```diff
@@ -199,6 +199,10 @@ impl Application {
         #[cfg(feature = "api")]
         let api_config = self.config.api;
 
+        // Any internal_logs sources will have grabbed a copy of the
+        // early buffer by this point and set up a subscriber.
+        crate::trace::stop_buffering();
```

Comment:
I was going to say we could get rid of this if we just `take` the buffer, but I'm assuming the primary purpose here is to prevent us from buffering forever if there is no source configured to read it.

Reply:
Exactly correct on the purpose.